tests/robustness/validate/watch.go

```diff
@@ -176,14 +176,10 @@ func validatePrevKV(t *testing.T, report report.ClientReport, history []model.Wa
 				if _, prevKeyExists := state.KeyValues[event.Key]; event.IsCreate == prevKeyExists {
 					t.Errorf("PrevKV - unexpected event ecountered, create event should not be in event history and update/delete event should be, event already exists: %t, is create event: %t, event: %+v", prevKeyExists, event.IsCreate, event)
 				}
-				var prevValue model.ValueRevision
-				if event.PrevValue != nil {
-					prevValue = *event.PrevValue
-				}
-				// This holds true when there is no compaction.
-				// Robustness tests currently do not do compaction.
-				if prevValue != state.KeyValues[event.Key] {
-					t.Errorf("PrevKV - PrevValue doesn't match previous value under the key %s, got: %+v, want: %+v", event.Key, prevValue, state.KeyValues[event.Key])
+				// Perform the check only if there is no compaction.
```

Comment:
Comment is not accurate. More accurate would be to say that we allow PrevValue to be nil as etcd doesn't guarantee its presence due to compaction.